src/components/input/input.html

```diff
@@ -35,7 +35,7 @@
 
       <label class="md-input-placeholder"
              [attr.for]="inputId"
-             [class.md-empty]="empty"
+             [class.md-empty]="empty && type !== 'date'"
```

Comment:
Any references? I guess in that cases there weren't any getters. If not the getter should probably not include that check.